spec/unit/facter/util/fact_gluster_peers_spec.rb

```diff
@@ -0,0 +1,89 @@
+# vim: syntax=ruby tabstop=2 softtabstop=2 shiftwidth=2
+
+require 'spec_helper'
+
+describe Facter::Util::Fact do
+  before { Facter.clear }
+
+  context 'gluster not running' do
+    before do
+      allow(Facter::Util::Resolution).to receive(:exec).with('gluster peer status --xml') {
+        'Connection failed. Please check if gluster daemon is operational.'
+      }
+    end
+    it {
+      expect(Facter.fact(:gluster_peers).value).to eq(nil)
+    }
+  end
+  context 'no peers defined' do
+    before do
+      allow(Facter::Util::Resolution).to receive(:exec).with('gluster peer status --xml') {
+        '<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+        <cliOutput>
+          <opRet>0</opRet>
+          <opErrno>0</opErrno>
+          <opErrstr/>
+          <peerStatus/>
+        </cliOutput>'
+      }
+    end
+
+    gluster_peers = {}
+
+    it {
+      expect(Facter.fact(:gluster_peers).value).to eq(gluster_peers)
+    }
+  end
+  context 'two peers defined' do
+    before do
+      allow(Facter::Util::Resolution).to receive(:exec).with('gluster peer status --xml') {
+        '<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+        <cliOutput>
+          <opRet>0</opRet>
+          <opErrno>0</opErrno>
+          <opErrstr/>
+          <peerStatus>
+            <peer>
+              <uuid>7d1148a2-f19e-4f18-818f-3396ddf38c30</uuid>
+              <hostname>peer2</hostname>
+              <hostnames>
+                <hostname>peer2</hostname>
+              </hostnames>
+              <connected>1</connected>
+              <state>3</state>
+              <stateStr>Peer in Cluster</stateStr>
+            </peer>
+            <peer>
+              <uuid>35f53c52-83dc-4100-a1f7-4a7cdeee074d</uuid>
+              <hostname>peer3</hostname>
+              <hostnames>
+                <hostname>peer3</hostname>
+              </hostnames>
+              <connected>1</connected>
+              <state>3</state>
+              <stateStr>Peer in Cluster</stateStr>
+            </peer>
+          </peerStatus>
+        </cliOutput>'
+      }
+    end
+
+    gluster_peers = {}
```

Comment:
You can specify hashes inline:
```ruby
{
  "peer2" => {
    "uuid" => "7d1148a2-f19e-4f18-818f-3396ddf38c30",
      "connected" => 1,
      "state" => 3,
      "status" => "Peer in Cluster",
  },
  "peer3" => {
    "uuid" => "35f53c52-83dc-4100-a1f7-4a7cdeee074d",
    "connected" => 1,
    "state" => 3,
    "status" => "Peer in Cluster",
  },
}
```

Possibly Rubocop has something about this, but the idea is clear I hope ;)